File: Initial Prototype/ServerSide/FindNDrive/FindNDriveServices2/DTOs/JourneyMessageRetrieverDTO.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="JourneyMessageRetrieverDTO.cs" company="">
//   
// </copyright>
// <summary>
//   Defines the JourneyMessageRetrieverDTO type.
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace FindNDriveServices2.DTOs
{
    using System.Runtime.Serialization;

    /// <summary>
    /// The journey message retriever dto.
    /// </summary>
    [DataContract]
    public class JourneyMessageRetrieverDTO
    {
        /// <summary>
        /// Gets or sets the journey id.
        /// </summary>
        [DataMember]
        public int JourneyId { get; set; }

        /// <summary>
        /// Gets or sets the user id.
        /// </summary>
        [DataMember]
        public int UserId { get; set; }

        /// <summary>
        /// Gets or sets the load range dto.
        /// </summary>
        [DataMember]
        public LoadRangeDTO LoadRangeDTO { get; set; }
    }
}

File: Initial Prototype/Client/MyApplicationProject/MyApplication/src/main/java/com/example/myapplication/dtos/JourneyMessageRetrieverDTO.java
package com.example.myapplication.dtos;

/**
 * DTO object used to retrieve a journey chat conversation history for a specific journey.
 */
public class JourneyMessageRetrieverDTO {
    public int JourneyId;
    public int UserId;
    LoadRangeDTO LoadRangeDTO;


    public JourneyMessageRetrieverDTO(int journeyId, int userId, LoadRangeDTO loadRangeDTO) {
        this.JourneyId = journeyId;
        this.UserId = userId;
        this.LoadRangeDTO = loadRangeDTO;
    }
}
